src/test/java/ExchangeRateFileReaderHandlerTest.java

```diff
@@ -0,0 +1,213 @@
+import org.currencygoldexchangeapp.datamodels.CurrencyExchange;
+import org.currencygoldexchangeapp.handlers.ExchangeRateFileReaderHandler;
+import org.currencygoldexchangeapp.services.CurrencyExchangeCalculateService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyDouble;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+
+public class ExchangeRateFileReaderHandlerTest {
+    @Mock
+    CurrencyExchangeCalculateService mockCalculateService;
+
+    @Test
+    public void readExchangeRates_withValidRecords() {
+        // Arrange
```

Comment:
Consider removing comments before the final production release to ensure clean code. However, at the moment they can be useful for you to understand what is going on  :) .

Reply:
imo it should help more other people understand what is going on, this is not in contradiction with clean architecture

Reply:
In my opinion, if the code is clearly written and self-explanatory, comments are often unnecessary but up to you.